test/runner/job_test.exs

```diff
@@ -45,4 +19,253 @@ defmodule ElixirBench.Runner.JobTest do
     assert job.context.worker_num_cores in 1..64
     assert job.context.worker_os in [:Linux, :Windows, :macOS]
   end
+
+  describe "start_job/6" do
+    test "run job and return results given execute function" do
+      task_fun = fn job ->
+        {:ok, %{job | log: "success", status: 0}}
+      end
+
+      {:ok, job} = Job.start_job(job_fixture(), task_fun: task_fun)
+
+      assert %{log: "success", status: 0} = job
+    end
+
+    test "not start job if there is other job running" do
+      task_fun = fn _job ->
+        :timer.sleep(2_000)
+        {:ok, %{}}
+      end
+
+      %{pid: pid} =
+        Task.Supervisor.async_nolink(ElixirBench.Runner.JobsSupervisor, fn ->
+          :timer.sleep(2_000)
+        end)
+
+      assert_raise MatchError, fn ->
+        Job.start_job(job_fixture(), task_fun: task_fun)
+      end
```

Comment:
I'm not sure how this works? :thinking: We only launch one job here, why should it fail? I guess it's in the supervisor above but at least to me that connection isn't clear

Reply:
in the line 41 I add another child process to the Supervisor that executes a sleep of 2 seconds, so when  Job.start_job spanws a new process to execute the benchmark it fails because there's already another child running.